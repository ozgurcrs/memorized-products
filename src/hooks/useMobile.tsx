import { useState, useEffect } from "react";

interface MobileHookResult {
  isMobile: boolean;
}

const MOBILE_WIDTH_THRESHOLD = 600;

const useMobile = (): MobileHookResult => {
  const [isMobile, setIsMobile] = useState<boolean>(false);

  useEffect(() => {
    const handleResize = () => {
      const currentWidth = window.innerWidth;
      setIsMobile(currentWidth <= MOBILE_WIDTH_THRESHOLD);
    };

    window.addEventListener("resize", handleResize);

    return () => {
      window.removeEventListener("resize", handleResize);
    };

    handleResize();
  }, []);

  return { isMobile };
};

export default useMobile;
