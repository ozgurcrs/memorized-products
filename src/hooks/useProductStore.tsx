import { useEffect, useMemo, useState } from "react";
import { ApiStatus, Product, Sort } from "../models";

export interface IUseProductStore {
  fetchProductApiStatus: ApiStatus;
  setSearchText: (query: string) => void;
  searchText: string;
  filterProductBySearch: Product[];
  setFilterOptions: (filterOptions: Sort) => void;
  filterOptions: Sort;
  getProductById: Function;
  isLoadingInfinityScroll: boolean;
  addProductToBasket: (product: Product) => void;
  productOfBasket: Product[];
  removeProductFromBasket: (product: Product) => void;
}

export const useProductStore = (): IUseProductStore => {
  const [fetchProductApiStatus, setFetchProductApiStatus] = useState<ApiStatus>(
    ApiStatus.IDLE
  );
  const [products, setProducts] = useState<Product[]>([]);
  const [pageNumber, setPageNumber] = useState<number>(1);
  const [searchText, setSearchText] = useState<string>("");
  const [productOfBasket, setProductOfBasket] = useState<Product[]>([]);
  const [filterOptions, setFilterOptions] = useState<Sort>(
    Sort.FromCheapToExpensive
  );
  const [isLoadingInfinityScroll, setIsLoadingInfinityScrol] =
    useState<boolean>(false);

  const getProducts = async () => {
    try {
      setFetchProductApiStatus(ApiStatus.LOADING);
      const { products } = await fetch(
        `https://teknasyon.netlify.app/.netlify/functions/products`,
        {
          headers: {
            "X-Access-Token": "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",
          },
        }
      ).then((res) => res.json());

      setFetchProductApiStatus(ApiStatus.SUCCESS);

      setProducts(products);
    } catch {
      setFetchProductApiStatus(ApiStatus.ERROR);
    }
  };

  const loadMoreProducts = () => {
    setPageNumber((prevPageNumber) => prevPageNumber + 1);
  };

  useEffect(() => {
    if (products.length) {
      let isFetching = false;
      let tempPageNumber = 1;

      const handleScroll = () => {
        const scrollable =
          document.documentElement.scrollHeight - window.innerHeight;
        const scrolled = window.scrollY;

        const threshold = 100;

        if (scrollable - scrolled < threshold && !isFetching) {
          const totalPageNumber = products.length / 10;

          if (tempPageNumber < totalPageNumber) {
            isFetching = true;
            setIsLoadingInfinityScrol(true);

            setTimeout(() => {
              loadMoreProducts();
              isFetching = false;
              setIsLoadingInfinityScrol(false);
              tempPageNumber = tempPageNumber + 1;
            }, 3000);
          }
        }
      };

      window.addEventListener("scroll", handleScroll);

      return () => {
        window.removeEventListener("scroll", handleScroll);
      };
    }
  }, [products]);

  const filteredProducts = useMemo(() => {
    return products.filter(
      (item) => item.title.toLowerCase().indexOf(searchText.toLowerCase()) > -1
    );
  }, [products, searchText]);

  const sortedProducts = useMemo(() => {
    if (Sort.FromCheapToExpensive === filterOptions) {
      return filteredProducts.sort(
        (a, b) => Number(a.variants[0].price) - Number(b.variants[0].price)
      );
    } else {
      return filteredProducts.sort(
        (a, b) => Number(b.variants[0].price) - Number(a.variants[0].price)
      );
    }
  }, [filteredProducts, filterOptions]);

  const getProductById = (id: string) => {
    if (products.length) {
      return products.find((item: Product) => String(item.id) === id);
    }
  };

  const addProductToBasket = (product: Product) => {
    setProductOfBasket((prev) => [...prev, product]);
  };

  const removeProductFromBasket = (product: Product) => {
    const { id } = product;

    const findProductIdx = productOfBasket.findIndex((item) => item.id === id);

    const updateProductList = [
      ...productOfBasket.slice(0, findProductIdx),
      ...productOfBasket.slice(findProductIdx + 1),
    ];

    setProductOfBasket(updateProductList);
  };

  useEffect(() => {
    getProducts();
  }, []);

  return {
    fetchProductApiStatus,
    setSearchText,
    filterProductBySearch: sortedProducts.filter(
      (_, index) => index < pageNumber * 10
    ),
    setFilterOptions,
    filterOptions,
    searchText,
    getProductById,
    isLoadingInfinityScroll,
    addProductToBasket,
    productOfBasket,
    removeProductFromBasket,
  };
};
