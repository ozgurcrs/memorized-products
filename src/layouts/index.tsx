import { FC, ReactNode } from "react";
import { Navigation } from "../components/Navigation";
import styles from "./layout.module.scss";

type Props = {
  children: ReactNode;
};

export const Layout: FC<Props> = ({ children }: { children: ReactNode }) => {
  return (
    <div className={styles["layout-container"]}>
      <Navigation />
      {children}
    </div>
  );
};
