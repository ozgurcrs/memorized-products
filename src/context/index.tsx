import { createContext } from "react";
import { useProductStore } from "../hooks/useProductStore";

export const ProductContext = createContext(
  {} as ReturnType<typeof useProductStore>
);
