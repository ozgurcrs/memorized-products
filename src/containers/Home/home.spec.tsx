import { Layout } from "../../layouts";
import { ApiStatus, Product, Sort } from "../../models";
import { ProductContext } from "../../context";
import { Home } from ".";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { IUseProductStore } from "../../hooks/useProductStore";

jest.mock("../../models");

const product: Product = {
  id: 7550808883419,
  title: "Axium 100 B93",
  body_html:
    '<p><em>This is a demonstration store. You can purchase products like this from <a href="//skiandscuba.com" target="_blank">The Ski Chalet &amp; Treasure Cove Scuba</a>.</em></p><ul>\n<li><span style="line-height: 1.4;">DIN 3/10</span></li>\n<li><span style="line-height: 1.4;">Toe lateral elasticity 45 mm</span></li>\n<li><span style="line-height: 1.4;">Glider AFC TEFLON</span></li>\n<li><span style="line-height: 1.4;">Toe piece DUAL ACTION II</span></li>\n<li><span style="line-height: 1.4;">Heel piece AXIUM2</span></li>\n<li><span style="line-height: 1.4;">Height interface 16 MM</span></li>\n<li><span style="line-height: 1.4;">Heel vertical elasticity 12 MM</span></li>\n<li><span style="line-height: 1.4;">Brake 2P 83-93 MM</span></li>\n<li><span style="line-height: 1.4;">Heel piece Adjustment 2P 24 MM</span></li>\n</ul>',
  vendor: "Rossignol",
  product_type: "Ski Bindings",
  created_at: "2022-02-23T16:19:32+03:00",
  handle: "rossignol-axium-100-b93-binding-2016",
  updated_at: "2022-02-23T16:28:20+03:00",
  published_at: "2022-02-23T16:19:31+03:00",
  template_suffix: null,
  status: "active",
  published_scope: "web",
  tags: "Ski Bindings",
  admin_graphql_api_id: "gid://shopify/Product/7550808883419",
  variants: [
    {
      id: 42509635551451,
      product_id: 7550808883419,
      title: "B93 / Black/White",
      price: "119.00",
      sku: "",
      position: 1,
      inventory_policy: "deny",
      compare_at_price: null,
      fulfillment_service: "manual",
      inventory_management: "shopify",
      option1: "B93",
      option2: "Black/White",
      option3: null,
      created_at: "2022-02-23T16:19:32+03:00",
      updated_at: "2022-02-23T16:19:32+03:00",
      taxable: true,
      barcode: "",
      grams: 2722,
      image_id: 36706928525531,
      weight: 6.001,
      weight_unit: "lb",
      inventory_item_id: 44603797242075,
      inventory_quantity: 4,
      old_inventory_quantity: 4,
      requires_shipping: true,
      admin_graphql_api_id: "gid://shopify/ProductVariant/42509635551451",
    },
  ],
  options: [
    {
      id: 9606279987419,
      product_id: 7550808883419,
      name: "Size",
      position: 1,
      values: ["B93"],
    },
    {
      id: 9606280020187,
      product_id: 7550808883419,
      name: "Color",
      position: 2,
      values: ["Black/White"],
    },
  ],
  images: [
    {
      id: 36706928525531,
      product_id: 7550808883419,
      position: 1,
      created_at: "2022-02-23T16:19:32+03:00",
      updated_at: "2022-02-23T16:19:32+03:00",
      alt: null,
      width: 554,
      height: 283,
      src: "https://cdn.shopify.com/s/files/1/0631/9122/0443/products/rossignol_RCDA028_AXIUM-100-B93-BLACK-WHITE.jpg?v=1645622372",
      variant_ids: [42509635551451],
      admin_graphql_api_id: "gid://shopify/ProductImage/36706928525531",
    },
  ],
  image: {
    id: 36706928525531,
    product_id: 7550808883419,
    position: 1,
    created_at: "2022-02-23T16:19:32+03:00",
    updated_at: "2022-02-23T16:19:32+03:00",
    alt: null,
    width: 554,
    height: 283,
    src: "https://cdn.shopify.com/s/files/1/0631/9122/0443/products/rossignol_RCDA028_AXIUM-100-B93-BLACK-WHITE.jpg?v=1645622372",
    variant_ids: [42509635551451],
    admin_graphql_api_id: "gid://shopify/ProductImage/36706928525531",
  },
};

const fakeProductStore = {
  fetchProductApiStatus: ApiStatus.SUCCESS,
  setSearchText: jest.fn(),
  searchText: "",
  filterProductBySearch: [product],
  setFilterOptions: jest.fn(),
  filterOptions: Sort.FromCheapToExpensive,
  getProductById: jest.fn(),
  isLoadingInfinityScroll: false,
  addProductToBasket: jest.fn(),
  productOfBasket: [product],
  removeProductFromBasket: jest.fn(),
};

const RenderProduct = () => {
  return (
    <ProductContext.Provider value={fakeProductStore as IUseProductStore}>
      <Layout>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Layout>
    </ProductContext.Provider>
  );
};

test("renders home page correctly", () => {
  const container = render(<RenderProduct />);
  expect(container).toMatchSnapshot();
});

test("renders image from card", () => {
  render(<RenderProduct />);

  const image = screen.getByTestId("image");

  expect(image).toBeDefined();
});

test("displays price on the screen", () => {
  render(<RenderProduct />);

  const price = screen.getByTestId("price");

  expect(price.textContent).toEqual("119.00â‚º");
});

test("adds product to the basket", async () => {
  render(<RenderProduct />);

  const addProduct = fakeProductStore.addProductToBasket;

  await waitFor(() => expect(addProduct).toHaveBeenCalledTimes(0));

  expect(fakeProductStore.productOfBasket.length).toBe(1);

  fireEvent.click(screen.getByTestId("add-product"));

  await waitFor(() => expect(addProduct).toHaveBeenCalledTimes(1));

  expect(fakeProductStore.productOfBasket.length).toBe(1);
});
