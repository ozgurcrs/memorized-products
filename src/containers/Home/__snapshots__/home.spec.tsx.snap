// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 7h-3V4c0-1.103-.897-2-2-2H9c-1.103 0-2 .897-2 2v3H4c-1.103 0-2 .897-2 2v6c0 1.103.897 2 2 2h3v3c0 1.103.897 2 2 2h6c1.103 0 2-.897 2-2v-3h3c1.103 0 2-.897 2-2V9c0-1.103-.897-2-2-2zm0 8h-5v4h.001v1H9v-5H4V9h5V4h6v5h5v6z"
              />
              <path
                d="M8 14v-4l-3 2zm8 0 3-2-3-2zm-6-6h4l-2-3zm2 11 2-3h-4z"
              />
              <circle
                cx="12"
                cy="12"
                r="2"
              />
            </svg>
            <h1>
              with
              <b>
                Product
              </b>
            </h1>
          </div>
          <div
            class="undefined "
          >
            <input
              placeholder="Search..."
              type="text"
            />
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
              />
            </svg>
          </div>
          <div>
            <span>
              1
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.071 1.243a.5.5 0 0 1 .858.514L3.383 6h9.234L10.07 1.757a.5.5 0 1 1 .858-.514L13.783 6H15.5a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5H15v5a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V9H.5a.5.5 0 0 1-.5-.5v-2A.5.5 0 0 1 .5 6h1.717L5.07 1.243zM3.5 10.5a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3z"
              />
            </svg>
          </div>
        </div>
        <article>
          <div>
            <div>
              <select
                name="filter"
              >
                <option
                  value="0"
                >
                  Lowest Price
                </option>
                <option
                  value="1"
                >
                  Highest Price
                </option>
              </select>
            </div>
            <div>
              <div>
                <a
                  href="/detail/7550808883419"
                >
                  <div>
                    <img
                      alt=""
                      data-testid="image"
                      src="https://cdn.shopify.com/s/files/1/0631/9122/0443/products/rossignol_RCDA028_AXIUM-100-B93-BLACK-WHITE.jpg?v=1645622372"
                    />
                  </div>
                  <div>
                    Axium 100 B93
                  </div>
                </a>
                <div>
                  <div />
                  <div
                    data-testid="price"
                  >
                    119.00
                    ₺
                  </div>
                  <div>
                    <div>
                      <span>
                        Size
                        :
                      </span>
                      <select
                        id=""
                        name=""
                      >
                        <option>
                          B93
                        </option>
                      </select>
                    </div>
                    <div>
                      <span>
                        Color
                        :
                      </span>
                      <select
                        id=""
                        name=""
                      >
                        <option>
                          Black/White
                        </option>
                      </select>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="add-product"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.071 1.243a.5.5 0 0 1 .858.514L3.383 6h9.234L10.07 1.757a.5.5 0 1 1 .858-.514L13.783 6H15.5a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5H15v5a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V9H.5a.5.5 0 0 1-.5-.5v-2A.5.5 0 0 1 .5 6h1.717L5.07 1.243zM3.5 10.5a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div />
          </div>
        </article>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 7h-3V4c0-1.103-.897-2-2-2H9c-1.103 0-2 .897-2 2v3H4c-1.103 0-2 .897-2 2v6c0 1.103.897 2 2 2h3v3c0 1.103.897 2 2 2h6c1.103 0 2-.897 2-2v-3h3c1.103 0 2-.897 2-2V9c0-1.103-.897-2-2-2zm0 8h-5v4h.001v1H9v-5H4V9h5V4h6v5h5v6z"
            />
            <path
              d="M8 14v-4l-3 2zm8 0 3-2-3-2zm-6-6h4l-2-3zm2 11 2-3h-4z"
            />
            <circle
              cx="12"
              cy="12"
              r="2"
            />
          </svg>
          <h1>
            with
            <b>
              Product
            </b>
          </h1>
        </div>
        <div
          class="undefined "
        >
          <input
            placeholder="Search..."
            type="text"
          />
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
            />
          </svg>
        </div>
        <div>
          <span>
            1
          </span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.071 1.243a.5.5 0 0 1 .858.514L3.383 6h9.234L10.07 1.757a.5.5 0 1 1 .858-.514L13.783 6H15.5a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5H15v5a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V9H.5a.5.5 0 0 1-.5-.5v-2A.5.5 0 0 1 .5 6h1.717L5.07 1.243zM3.5 10.5a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3z"
            />
          </svg>
        </div>
      </div>
      <article>
        <div>
          <div>
            <select
              name="filter"
            >
              <option
                value="0"
              >
                Lowest Price
              </option>
              <option
                value="1"
              >
                Highest Price
              </option>
            </select>
          </div>
          <div>
            <div>
              <a
                href="/detail/7550808883419"
              >
                <div>
                  <img
                    alt=""
                    data-testid="image"
                    src="https://cdn.shopify.com/s/files/1/0631/9122/0443/products/rossignol_RCDA028_AXIUM-100-B93-BLACK-WHITE.jpg?v=1645622372"
                  />
                </div>
                <div>
                  Axium 100 B93
                </div>
              </a>
              <div>
                <div />
                <div
                  data-testid="price"
                >
                  119.00
                  ₺
                </div>
                <div>
                  <div>
                    <span>
                      Size
                      :
                    </span>
                    <select
                      id=""
                      name=""
                    >
                      <option>
                        B93
                      </option>
                    </select>
                  </div>
                  <div>
                    <span>
                      Color
                      :
                    </span>
                    <select
                      id=""
                      name=""
                    >
                      <option>
                        Black/White
                      </option>
                    </select>
                  </div>
                </div>
              </div>
              <div
                data-testid="add-product"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.071 1.243a.5.5 0 0 1 .858.514L3.383 6h9.234L10.07 1.757a.5.5 0 1 1 .858-.514L13.783 6H15.5a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5H15v5a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V9H.5a.5.5 0 0 1-.5-.5v-2A.5.5 0 0 1 .5 6h1.717L5.07 1.243zM3.5 10.5a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3zm2.5 0a.5.5 0 1 0-1 0v3a.5.5 0 0 0 1 0v-3z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div />
        </div>
      </article>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
